core/src/main/java/com/datastax/oss/driver/internal/core/control/ControlConnection.java

```diff
@@ -273,12 +275,28 @@ private void init(boolean listenToClusterEvents, boolean reconnectOnFailure) {
               firstConnectionAttemptFuture.complete(null);
             },
             error -> {
-              if (reconnectOnFailure && !closeWasCalled) {
+              boolean authFailure = true;
```

Comment:
we are assuming upfront that error is an `authFailure`? 
I think that `if (error instanceof AllNodesFailedException) {}` is `false` it does not mean that it is an `authFailure`, or is it?